packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -374,27 +374,26 @@ - (void)disableContextMenu
 - (void)handleLongPress:(UILongPressGestureRecognizer *)gesture
 {
 #if !TARGET_OS_UIKITFORMAC
-  if (@available(iOS 16.0, *)) {
-    CGPoint location = [gesture locationInView:self];
-    UIEditMenuConfiguration *config = [UIEditMenuConfiguration configurationWithIdentifier:nil sourcePoint:location];
-    if (_editMenuInteraction) {
-      [_editMenuInteraction presentEditMenuWithConfiguration:config];
-    }
-    return;
-  }
-  // TODO: Adopt showMenuFromRect (necessary for UIKitForMac)
-  UIMenuController *menuController = [UIMenuController sharedMenuController];
-
-  if (menuController.isMenuVisible) {
-    return;
-  }
-
-  if (!self.isFirstResponder) {
-    [self becomeFirstResponder];
+  if (@available(iOS 16.0, macCatalyst 16.0, *)) { // [visionOS]
+	CGPoint location = [gesture locationInView:self];
+	UIEditMenuConfiguration *config = [UIEditMenuConfiguration configurationWithIdentifier:nil sourcePoint:location];
+	if (_editMenuInteraction) {
+	  [_editMenuInteraction presentEditMenuWithConfiguration:config];
+	  // [visionOS
+	} else {
+	  UIMenuController *menuController = [UIMenuController sharedMenuController];
+
+	  if (menuController.isMenuVisible) {
+		return;
+	  }
+
+	  if (!self.isFirstResponder) {
+		[self becomeFirstResponder];
+	  }
```

Comment:
Is this still needed? I don't see it upstream: https://github.com/facebook/react-native/blob/5f75e9b90d4f998403101ae92924778df31d36fb/packages/react-native/Libraries/Text/Text/RCTTextView.mm#L250-L256

Reply:
Looks like my change upstream removed it, but I dont' think that was actually intentional.. I should probably add it back. https://github.com/facebook/react-native/pull/42277

Reply:
I cherry-picked https://github.com/facebook/react-native/commit/1dd647484aff00a079923a5c0e763e3f54ccc210 and https://github.com/facebook/react-native/commit/6801fc3f9bf172f5ca929127576558be04e3f99c into this PR as earlier commits. Therefore the last commit "Add native changes for visionOS" should be cleaner.

For the removed block, I pinged my original PR reviewer whether it should be added back. If so,  I'll make an extra cherry-pick to do so.